pinot-query-planner/src/main/java/org/apache/pinot/query/planner/physical/DispatchablePlanContext.java

```diff
@@ -71,17 +88,32 @@ public long getRequestId() {
     return _requestId;
   }
 
+  public PlannerContext getPlannerContext() {
+    return _plannerContext;
+  }
+
+  public PairList<Integer, String> getResultFields() {
+    return _resultFields;
+  }
+
   // Returns all the table names.
   public Set<String> getTableNames() {
     return _tableNames;
   }
 
-  public PairList<Integer, String> getResultFields() {
-    return _resultFields;
+  /// Tracks non-lookup tables queried in leaf stages, which are used to determine the servers to use for intermediate
+  /// stages. Returns `null` if leaf servers are directly used for intermediate stages.
+  @Nullable
+  public Set<String> getNonLookupTables() {
+    return _nonLookupTables;
```

Comment:
I think it is unexpected that a getter method like this one returns null under some distribution conditions. Shouln't that be another getter method to be called?

The current semantic seems error prone